test/functional/job/job_spec.lua

```diff
@@ -37,6 +37,23 @@ describe('jobs', function()
     eq({'notification', 'exit', {0, 0}}, next_msg())
   end)
 
+  it('changes to the provided cwd (/)', function()
+    nvim('command', "let g:job_opts.cwd = '/'")
+    nvim('command', "let j = jobstart('pwd', g:job_opts)")
+    eq({'notification', 'stdout', {0, {'/', ''}}}, next_msg())
+    eq({'notification', 'exit', {0, 0}}, next_msg())
+  end)
+
+  it('changes to the provided cwd (/tmp/nvim.XXXXX)', function()
+    dir = '/tmp/nvim.' .. tostring(math.random(10000,99999))
```

Comment:
Vim's `tempname()` is more portable (will work on Windows too; though you didn't add support to `pty_process_win.c`, the failing test will remind us to add it later)